ivadomed/losses.py

```diff
@@ -405,6 +405,47 @@ def forward(self, input, target):
 
         return mean_loss
 
+class JointsMSELoss(nn.Module):
+    """
+    Joint MSE loss for pose estimation method.
+    .. seealso::
+        Alejandro Newell et al. "Stacked Hourglass Networks for Human Pose Estimation."
+        Proceedings of the European Conference on Computer Vision. 2016.
+    Args:
+        output (tensor): prediction mask mask
+        target_and_weights (list): list as follows:
+          -- ground truth mask (Tensor): estimated mask by the pose model
+          -- target_weight (Tensor): visibility of the intervertebral disc to control the loss value for the training process
```

Comment:
The loss function is not limited to HourglassNet and can be applied to any estimation method with any particular number of output channels. This loss function applies the visibility vector to control the loss value and only focus on the real targets. It can be switched off using `use_target_weight` flag.

Reply:
> The loss function is not limited to HourglassNet and can be applied to any estimation method with any particular number of output channels.

Ah, I see! :+1: 

> It can be switched off using `use_target_weight` flag.

Even if `use_target_weight` is turned set to `False`, `targets_and_weights` will still require a list of size 2, though:

https://github.com/ivadomed/ivadomed/blob/a7211e5638d326d290321b1d9f32242298c86d62/ivadomed/losses.py#L429

So, if a single argument is passed (e.g. just the target), it will fail:

```python
>>> target_only = np.array([1, 2, 3, 4, 5])
>>> target, target_weight = target_only
Traceback (most recent call last):
  File "<input>", line 1, in <module>
ValueError: too many values to unpack (expected 2)
```

Reply:
Well, why you should pass only one argument to this function? It is obvious that it will generate an error, to avoid conditions on use_target_weight should be checked to unpack the target and weight.

Reply:
Sorry, I mispoke. What I meant was -- 2 arguments instead of 3.

Reply:
> to avoid conditions on use_target_weight should be checked to unpack the target and weight.

I'm not sure I understand the wording here. Do you mean to say that you will check `target_and_weights` to see if it is a list before unpacking it?

Reply:
Yes, something like :
`target, target_weight = target_and_weights if self.use_target_weight else (target_and_weights, None)`

Reply:
One other idea could be to get rid of `self.use_target_weight`, and infer the usage based on the input. Something like:

```python
if isinstance(input, list) and len(input) = 2 and all(isinstance(n, torch.Tensor) for n in input):
    target, weights = input
elif isinstance(input, torch.Tensor):
    target, weights = input, None
else:
    raise ValueError("Input must either a Tensor (target) or a list of 2 Tensors (target, weights).")
```

Reply:
updated.